node/trieIterators/factory/sovereignDirectStakedListHandlerFactory_test.go

```diff
@@ -0,0 +1,41 @@
+package factory_test
+
+import (
+	"fmt"
+	"github.com/multiversx/mx-chain-go/node/trieIterators"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/node/mock"
+	trieIteratorsFactory "github.com/multiversx/mx-chain-go/node/trieIterators/factory"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	stateMock "github.com/multiversx/mx-chain-go/testscommon/state"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSovereignDirectStakedListHandlerFactory(t *testing.T) {
+	t.Parallel()
+
+	directStakedListHandlerFactory := trieIteratorsFactory.NewDirectStakedListHandlerFactory()
```

Comment:
Wrong copy pasted code. Should use constructor for sovereign.
Please also check the factory type to be of type sovereign